### ISSUE TITLE
[Go target] Generated parser contains interfaces without requisite methods for tree navigation

### ISSUE THREAD

**Issue by kaby76** [antlr/antlr4#3926]
This is a placeholder for a problem someone noted in StackOverflow ([so74022012](https://stackoverflow.com/q/74022012/4779853). It might be somewhere in the antlr/antlr4 issues, but I cannot find it.

The Go target doesn't generate code with accessor functions using the RHS symbol name as is done with other targets. Consequently, code that makes navigations using the accessor functions have to be type casted in order to go from one node to another in the parse tree. 

#### Grammar
[abb](https://github.com/antlr/grammars-v4/blob/master/abb/abbParser.g4)

```
module  : moduleData EOF   ;
moduleData : MODULE moduleName NEWLINE  dataList  NEWLINE* ENDMODULE ;
```

#### Generated Java
```
public static class ModuleContext extends ParserRuleContext {
		public ModuleDataContext moduleData() { ... }
}
public static class ModuleDataContext extends ParserRuleContext {
		public TerminalNode MODULE() { return getToken(abbParser.MODULE, 0); }
		public ModuleNameContext moduleName() { ... }
}
```
##### Navigation in Java
```
module.moduleData().moduleName() ...
```

#### Generated Go
```
func (s *ModuleContext) ModuleData() IModuleDataContext { ... }
type IModuleDataContext interface {
	antlr.ParserRuleContext

	// GetParser returns the parser.
	GetParser() antlr.Parser

	// IsModuleDataContext differentiates from other interfaces.
	IsModuleDataContext()
}
type ModuleDataContext struct {
	*antlr.BaseParserRuleContext
	parser antlr.Parser
}
```
##### Navigation in Go
```
module.(*parser.ModuleContext).moduleData().(*parser.ModuleDataContext).moduleName()
```
In the generated Go, the accessors return interfaces (unlike that in Java or C#), and the interfaces do not have methods declared for the accessors contained in that interface.

**Comment by j3r3miah** [comment]
Hey @kaby76, thanks for the quick response to my StackOverflow post! What are your thoughts on this attempt to fix this issue? https://github.com/antlr/antlr4/pull/3927

**Comment by kaby76** [comment]
@j3r3miah Yes, that is exactly what needs to be generated for the interface. @jimidle can help out if needed.

**Comment by jimidle** [comment]
Let me take a look. Generally I have been using the Listener, which works well. I have not really explored the tree walker. It seems like maybe it was left behind a little.

**Comment by j3r3miah** [comment]
Hi @jimidle @kaby76, can you please review https://github.com/antlr/antlr4/pull/3927?

**Comment by jimidle** [comment]
@parrt The PR to fix this has been merged and it checks out. We can close this issue now.
